python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,151 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any, List
+
+from revng.model import Binary, get_element_by_path, make_diff
+from revng.tupletree import TypedList
+
+
+class Project(ABC):
+    def __init__(self) -> None:
+        self.model: Binary = Binary()
+        self.last_saved_model: Binary = deepcopy(Binary)
+        self.pipeline_description: dict = {}
+        self.input_binary: str = ""
+
+    @abstractmethod
+    def import_binary(self, input_binary: str):
+        pass
+
+    @abstractmethod
+    def get_artifact(self, artifact_name: str) -> bytes | str:
+        """
+        High level method that the user can call to fetch an artifact without targets.
+        """
+        pass
+
+    @abstractmethod
+    def _get_artifact(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
+        """
+        Resolves the target logic, performs MIME conversion and dispatches
+        the work to the `_get_artifact_internal` to actually get the artifact.
+        """
+        pass
+
+    @abstractmethod
+    def _get_artifact_internal(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
+        """
+        The actual implementation of how to fetch te artifacts, eg: call subprocess or gql.
+        """
+        pass
```

Comment:
The actual implementation of these two functions does the exact opposite as what's stated in the doc: `_get_artifact_internal` calls `_get_artifact`, swap the two functions names in the docs.